packages/flutter/test/material/time_test.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('hourFormat', () {
+    tearDown(() {
+      ui.window.debugOverrideAlwaysUse24HourFormat(false);
+    });
+
+    testWidgets('respects window.alwaysUse24HourFormat', (WidgetTester tester) {
+      expect(hourFormat(of: TimeOfDayFormat.a_space_h_colon_mm), HourFormat.h);
+      ui.window.debugOverrideAlwaysUse24HourFormat(true);
+      expect(hourFormat(of: TimeOfDayFormat.a_space_h_colon_mm), HourFormat.HH);
+      ui.window.debugOverrideAlwaysUse24HourFormat(false);
+      expect(hourFormat(of: TimeOfDayFormat.a_space_h_colon_mm), HourFormat.h);
+    });
```

Comment:
you should also check that changing the state of the flag changes the actual rendered widget